enterprise/backend/test/metabase_enterprise/task/cache_test.clj

```diff
@@ -468,97 +475,163 @@
+                    (finally
+                      (delete-cache-entry! original-cache-entry))))))))))))
 
 (deftest refresh-duration-cache-card-e2e-test
   (mt/with-premium-features #{:cache-granular-controls :cache-preemptive}
-    (testing "Do we successfully execute a refresh query for a :duration cache config on a card?"
+    (testing "Do we successfully execute a refresh query for a :duration cache config on a card?\n"
       (binding [qp.util/*execute-async?*             false
                 task.cache/*run-cache-refresh-async* false]
-        (mt/with-temp [:model/Card {card-id :id} {:name "Cached card"
-                                                  :dataset_query (parameterized-native-query)}
-                       :model/CacheConfig _ {:model "question"
-                                             :model_id card-id
-                                             :strategy :duration
-                                             :refresh_automatically true
-                                             :config {:unit "hours" :duration 1}}]
-          ;; Run card once to populate cache
-          (is (= [[1000]] (mt/rows (run-query-for-card-id card-id []))))
-          ;; Run again to register a cache hit
-          (is (= [[1000]] (mt/rows (run-query-for-card-id card-id []))))
-          (let [cache-timestamp-1 (:updated_at (most-recent-cache-entry))]
-            ;; Manually expire the existing cache entry. Now the cache should be refreshed.
-            (expire-most-recent-cache-entry!)
-            (@#'task.cache/refresh-cache-configs!)
-            (let [cache-timestamp-2 (:updated_at (most-recent-cache-entry))]
-              (is (t/before? cache-timestamp-1 cache-timestamp-2)))))))))
+        (doseq [{:keys [label query parameters result]} (testing-queries)]
+          (testing (format "\nTesting with %s:" label)
+            (mt/with-temp [:model/Card {card-id :id} {:name "Cached card"
+                                                      :dataset_query query}
+                           :model/CacheConfig _ {:model "question"
+                                                 :model_id card-id
+                                                 :strategy :duration
+                                                 :refresh_automatically true
+                                                 :config {:unit "hours" :duration 1}}]
+              ;; Run card once to populate cache
+              (let [original-result (run-query-for-card-id card-id parameters)]
+                (def original-result original-result)
+                (is (= result (mt/rows original-result)))
+                ;; Run again to register a cache hit
+                (is (= result (mt/rows (run-query-for-card-id card-id parameters))))
+                (let [original-cache-entry (most-recent-cache-entry)]
+                  (try
+                    ;; Manually expire the existing cache entry. Now the cache should be refreshable
```

Comment:
Would putting in a call to `(@#'task.cache/refresh-cache-configs!)` and asserting that it is 0 make sense here?